applications/ParticleMechanicsApplication/custom_utilities/mpm_boundary_rotation_utility.h

```diff
@@ -179,26 +179,10 @@ class MPMBoundaryRotationUtility: public CoordinateTransformationUtils<TLocalMat
 	void ApplySlipCondition(TLocalVectorType& rLocalVector,
 			GeometryType& rGeometry) const override
 	{
-		if (rLocalVector.size() > 0)
-		{
-			for(unsigned int itNode = 0; itNode < rGeometry.PointsNumber(); ++itNode)
-			{
-				if( this->IsSlip(rGeometry[itNode]) )
-				{
-					// We fix the first momentum dof (normal component) for each rotated block
-					unsigned int j = itNode * this->GetBlockSize(); // +1
-
-					// Get the displacement of the boundary mesh, this does not assume that the mesh is moving.
-					// If the mesh is moving, need to consider the displacement of the moving mesh into account.
-					const array_1d<double,3> & displacement = rGeometry[itNode].FastGetSolutionStepValue(DISPLACEMENT);
-					array_1d<double,3> rN = rGeometry[itNode].FastGetSolutionStepValue(NORMAL);
-					this->Normalize(rN);
-
-					rLocalVector[j] = inner_prod(rN,displacement);
-
-				}
-			}
-		}
+        // creates an empty dummy matrix to pass into the 'full' ApplySlipCondition -- this dummy matrix is
+        // ignored, effectively only updating the RHS
+        TLocalMatrixType dummyMatrix = *(new TLocalMatrixType());
```

Comment:
This is a memory leak, please do not allocate with new

Reply:
Oh wait sorry I see what you mean and where the problem lies now: I should just change it to
`TLocalMatrixType dummyMatrix = TLocalMatrixType();`

Thanks for the feedback!

Reply:
I've made the changes to both occurrences in the latest commit d7ad7d57dc4cb8a83ecf73410c8dea4a9c4be5f3